### DISCUSSION TITLE
Uploaded images no longer displaying

### DISCUSSION THREAD

**Discussion by voron69-bit** [sysadminsmedia/homebox#248] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/7b2c0450-fbcc-4498-85ec-86e09752c5bc)
![image](https://github.com/user-attachments/assets/81381d55-bbb3-4c71-ad35-5cb6db592ae4)
I took a long time uploading stock photos of all my objects in a tote. They were all working, and looked great, a couple days later, all the image links are broken and don't work. I checked the server, and all the files are there in the appdata/HomeBox/Documents/ directory, but they are all renamed is system generated names like "ff86fb77-8103-459e-a7d0-4c02d57dc0a4.jpg".

These were uploaded stock photos from amazon. Not that it should matter, and not taken from a camera. Though all the ones I created with a camera photo are also not working. 

What (if anything) have I done wrong here?

**Comment by voron69-bit** [comment]
This is when accessing the site either from local network, or external cname URL. As a matter of fact, I just did about 30 more objects using my phone, and they go in, once the page is closed and reopened, they all have the broken link.

**Reply by katosdev** [reply]
Hey @voron69-bit ! 
Sorry to hear of the issues that you are having. Is there anything in the browser console that might indicate any errors?

**Reply by voron69-bit** [reply]
Anything specific I can get for you?

**Reply by katosdev** [reply]
> Anything specific I can get for you?

Really just looking for any errors or warnings in the console at this point to be honest with you, and then can narrow down from there.

**Reply by mcarbonne** [reply]
You can also check that images are still present in your data volume:
```bash
$ tree  /your-path/to/homebox-data
/your-path/to/homebox-data
├── 28824c84-2740-4815-9d1b-62641713283e
│   └── documents
│       ├── 307d33c2-05b1-498e-bfea-f62c5c64371f.jpg
│       ├── a856b534-e6d0-4e1a-8fb8-faf6c5968933.jpg
[....]
│       └── 1b70a4c9-2636-478e-840a-84d8b52fdb05.jpg
├── homebox.db
├── homebox.db-shm
└── homebox.db-wal
```

**Reply by mcarbonne** [reply]
Also, if you are using firefox, try "Ctrl+Shift+R" (they are lot of issues with cache and nuxt applications :/)
And finally, you can try to browse you database (`sqlitebrowser homebox.db`) and check that the attachments are still there and pointing to correct files (table `attachments`)

**Reply by voron69-bit** [reply]
> You can also check that images are still present in your data volume:
> 
> ```shell
> $ tree  /your-path/to/homebox-data
> /your-path/to/homebox-data
> ├── 28824c84-2740-4815-9d1b-62641713283e
> │   └── documents
> │       ├── 307d33c2-05b1-498e-bfea-f62c5c64371f.jpg
> │       ├── a856b534-e6d0-4e1a-8fb8-faf6c5968933.jpg
> [....]
> │       └── 1b70a4c9-2636-478e-840a-84d8b52fdb05.jpg
> ├── homebox.db
> ├── homebox.db-shm
> └── homebox.db-wal
> ```

That's exactly what I see in my data volume. All the images are there. I use Brave browser, and I've tried Ctrl+Shift+R, and Shift+Refresh, nothing gets it back.

Though I did just try Chrome, and it works fine. Any way we can fix this? I use Brave exclusively on desktop and mobile.
Also seems to work in Private Brave browser as well.

**Reply by katosdev** [reply]
Sounds to me like an addin of it works in private.
Do you have an adblocker running? Please try disabling that and see if any difference @voron69-bit

**Reply by voron69-bit** [reply]
> Sounds to me like an addin of it works in private. Do you have an adblocker running? Please try disabling that and see if any difference @voron69-bit

Disabled the Brave "Shields up" which didn't effect it, then Ublock Origin. Still no change. That is the only extension I'm running. Though it WAS working previously on Brave.

**Reply by katosdev** [reply]
Let me do some testing for you on brave today and see if I can replicate at all

**Reply by voron69-bit** [reply]
> Let me do some testing for you on brave today and see if I can replicate at all

That'd be awesome! Let me know if there is anything I can get you.

**Reply by katosdev** [reply]
> > Let me do some testing for you on brave today and see if I can replicate at all
> 
> That'd be awesome! Let me know if there is anything I can get you.

I've been unable to replicate this, can you share details of your Brave configuration? Extensions etc please.

**Reply by tankerkiller125** [reply]
I spent part of my day today trying to sort this out myself, and no matter what I did to try and replicate it, I was unable to do so. I'm truly at a lose for how one browser can't get images but others can. And for the same exact person at that.

**Reply by voron69-bit** [reply, marked as answer]
I deleted all cache for all my browsers, both pc and phone, and its working normally now. How? I have no idea.

**Reply by voron69-bit** [reply]
@tankerkiller125 @katosdev 

So I just wanted to let you guys know, I've had to do this about 4 times now, on both mobile and desktop. It just happened again after updating to the most recent version. All images are broken links, Until I go into the site data and clear all the data. This is clearly a bug, and needs to be resolved.

**Reply by katosdev** [reply]
Very strange, I can’t replicate this one at all. 
Are you able to open as an issue for us please and include some details on your hosting configuration and client browsers so that we can try to narrow this down?

Apologies for the inconvenience

**Reply by tummydummy** [reply]
This "broken image" issue happens to me quite a lot on Brave Browser. Definitley a bug.
Brave 1.80.115 (Official Build) (64-bit)
Chromium: 138.0.7204.97

**Reply by Ethan1023** [reply]
I just got this - was fixed by clearing browser data.

Just in case it helps narrowing down the cause, this is my setup:

Firefox Version 134.0 (64-bit), Linux

Homebox running in a docker compose using a cifs share as backend
```
services:
  homebox:
    image: ghcr.io/sysadminsmedia/homebox:latest
    container_name: homebox
    restart: always
    environment:
    - HBOX_LOG_LEVEL=info
    - HBOX_LOG_FORMAT=text
    - HBOX_WEB_MAX_FILE_UPLOAD=10
    # Please consider allowing analytics to help us improve Homebox (basic computer information, no personal data)
    - HBOX_OPTIONS_ALLOW_ANALYTICS=false
    volumes:
      - /mnt/smb/homebox:/data/
    ports:
      - 3100:7745
```
[Version: v0.21.0 Build: 27e9eb2277b90783f6d3a03d4895ce7dec6f0fe9 ](https://github.com/sysadminsmedia/homebox/releases/tag/v0.21.0)

Also I have 249 items and 92 file attachments (from looking directly at the `documents` folder).

Would like to add that images that I had just uploaded (after images stopped rendering) also didn't render. When I clicked the download button it downloaded an empty html with name `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.html` (hex digits, I deleted it and firefox isn't letting me copy the name from the download list :|)

Is there an issue for this?

**Reply by voron69-bit** [reply]
> Very strange, I can’t replicate this one at all. Are you able to open as an issue for us please and include some details on your hosting configuration and client browsers so that we can try to narrow this down?
> 
> Apologies for the inconvenience

I switched to Firefox. A lot of self hosted apps have issues with BRave, and it wasn't worth me hanging on it. No issues with Homebox since.

**Reply by Ethan1023** [reply]
Odd cos I just got it on Firefox. If I get it again I'll try get more details and create and issue for them to look at, but not a huge hassle to reset the local storage. So as long as it doesn't happen too often it's not very annoying.

**Comment by katosdev** [comment]
Fixed in V0.22.3
